### DISCUSSION TITLE
API task - 'State of this instance has been externally set to None'

### DISCUSSION THREAD

**Discussion by nicolearugay** [apache/airflow#30211] [General]
### Apache Airflow version

2.5.2

### What happened

- LocalExecutor
- MySQL backend
- macOS Big Sur Version 11.6 M1 chip

I'm trying to run a DAG locally and manually in the webserver UI and one of my tasks hang when I run a GET request to an API. The task continues 'running' until I get the error message below. 

### What you think should happen instead

The task should run fast and completely. 

```
[2023-03-19T15:43:46.011-0400] {logging_mixin.py:137} INFO - Before get.
[2023-03-19T15:45:26.748-0400] {local_task_job.py:272} WARNING - State of this instance has been externally set to None. Terminating instance.
[2023-03-19T15:45:26.755-0400] {process_utils.py:129} INFO - Sending Signals.SIGTERM to group 9063. PIDs of all processes in the group: [9063]
[2023-03-19T15:45:26.755-0400] {process_utils.py:84} INFO - Sending the signal Signals.SIGTERM to group 9063
[2023-03-19T15:46:26.758-0400] {process_utils.py:147} WARNING - process psutil.Process(pid=9063, name='Python', status='running', started='15:43:45') did not respond to SIGTERM. Trying SIGKILL
[2023-03-19T15:46:26.765-0400] {process_utils.py:84} INFO - Sending the signal Signals.SIGKILL to group 9063
[2023-03-19T15:46:26.780-0400] {process_utils.py:79} INFO - Process psutil.Process(pid=9063, name='Python', status='terminated', exitcode=<Negsignal.SIGKILL: -9>, started='15:43:45') (9063) terminated with exit code Negsignal.SIGKILL
[2023-03-19T15:46:26.781-0400] {standard_task_runner.py:168} ERROR - Job 4 was killed before it finished (likely due to running out of memory)
```

### How to reproduce

```
from airflow import DAG
from datetime import timedelta, datetime
from airflow.operators.python import PythonOperator

import requests
import logging
import json
from datetime import datetime 

default_args = {
        'owner' : 'airflow',
        'start_date' : datetime(2023, 3, 10),

}

def notify():
       url = f"https://www.google.com"
       print("Before get.")
       logging.basicConfig(level=logging.DEBUG)
       s = requests.Session()
       response = s.get(url)
       print(response)


with DAG(dag_id='weather_dag_5', default_args=default_args, schedule_interval='@daily', catchup=False):
        PythonOperator(
        task_id="notify",
        python_callable=notify
        )

notify
```

### Operating System

macOS Big Sur Version 11.6 M1

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by hussein-awala** [comment]
I cannot reproduce it on amd64, it may be a problem with arm processor (with one of the depenedencies), knowing that the support of arm arch is still experimental in Airflow ([doc](https://airflow.apache.org/docs/docker-stack/index.html#support)).

Can you export the list of the installed packages?
```bash
pip freeze
```

**Comment by potiuk** [comment]
I believe you might have SSL issues in ARM. There are a number of problems with SSL in ARM image that lead to segmentation violaion - requiring NO_PROXY env variable for example. See those thread for possible workarounds https://github.com/apache/airflow/discussions?discussions_q=Negsignal.SIGSEGV. You do not have SEGV in your logs, but it is likely it is somewhat hidden in your case so you should try it  out.

This migh also be a problem with networking badly configured on your machine and your DNS query hanging forever without the possibility of process being terminated. I'd look at that in your case. 

It also might be that for some reason that due to the non-interactive shell that airflow task runs in (https://www.geeksforgeeks.org/shell-scripting-interactive-and-non-interactive-shell/) is unable to display some prompt and get user input.

In any case you should be able to reproduce the event by making sure you set the exact same environment your airflow task runs in and trying to manually run the same python command - that should reproduce the behaviour that you should be able to diagnose and fix.

**Comment by nicolearugay** [comment]
Thanks for your help. I was able to get my DAG to run as expected when using Docker.
